### PR TITLE
Use a dedicated network for node/node traffic

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

File: fragments/master-ansible.sh [hunk 1]
```diff
@@ -126,4 +126,11 @@ export OS_REGION_NAME=$os_region_name
 ansible-playbook --inventory /var/lib/ansible/inventory \
     /var/lib/ansible/playbooks/main.yml > /var/log/ansible.$$ 2>&1
 
+ansible all \
+    -m shell \
+    -a 'iptables -A DOCKER -p tcp -j ACCEPT' \
+    -u cloud-user --sudo \
+    -i /var/lib/ansible/inventory \
+    >> /var/log/ansible-post-$$.log 2>&1
+
```
Comment (jprovaznik) [inline fragments/master-ansible.sh]: this can be removed (it's included in main.yaml playbook)

File: master.yaml [hunk 2]
```diff
@@ -298,6 +321,12 @@ resources:
         - path: /usr/local/bin/retry
           permissions: 0755
           content: {get_file: fragments/retry.sh}
+        - path: /etc/sysconfig/network-scripts/ifcfg-eth1
+          content:
+            str_replace:
+              params:
+                $IFNAME: eth1
```
Comment (jprovaznik) [inline master.yaml:328]: I wonder if there is a reason to make this parametrized if eth1 is hardcoded on other places anyway?

Comment (lebauce) [inline master.yaml:328]: I did so in the - very improbable - case where we would need to use it for an other interface. I can change it if you want

File: fragments/master-ansible.sh (hunk 1 above)
Comment (lebauce) [inline fragments/master-ansible.sh]: Done. Thanks

**Comment by jprovaznik** [comment]
LGTM, stack deployment finishes successfully, but I'm hitting following errors when runnin "oadm diagnostics" on the deployed master node (3 masters, 2 nodes):

```
  ERROR: [DP1016 from diagnostic PodCheckAuth@openshift/origin/pkg/diagnostics/pod/auth.go:113]
          DNS resolution for registry address docker-registry.default.svc.cluster.local returned an error; container DNS is likely incorrect. The error was: read udp 172.17.0.2:54057->172.30.0.1:53: read: no route to host

   [Note] Running diagnostic: PodCheckDns
          Description: Check that DNS within a pod works as expected

   ERROR: [DP2003 from diagnostic PodCheckDns@openshift/origin/pkg/diagnostics/pod/dns.go:72]
          The first /etc/resolv.conf nameserver 172.30.0.1
          could not resolve kubernetes.default.svc.cluster.local.
          Error: read udp 172.17.0.2:35634->172.30.0.1:53: read: no route to host
          This nameserver points to the master's SkyDNS which is critical for
          resolving cluster names, e.g. for Services.
```

I didn't hit these w/o the patch, could you please check on your setup if you hit this?

**Comment by jprovaznik** [comment]
actually it might be unrelated - these errors don't occur depending on which node the checking pod is deployed

**Comment by jprovaznik** [comment]
I didn't hit the error above w/o this patch (on any node), could you please confirm?

**Comment by jprovaznik** [comment]
One more issue I started hitting today is probably related to flannel too, ansible fails during "PLAY [Configure flannel etcd certificates] ***********************************\* " with error:
TASK: [etcd_certificates | Ensure generated_certs directory present] *********\* 
fatal: [flannel-origin-master-0.example.com] => Failed to template {{ etcd_needing_client_certs | default([]) }}: Failed to template {{ g_master_hosts | union(g_node_hosts) | union(g_etcd_hosts) | union(g_lb_hosts) | union(g_nfs_hosts) | union(g_new_node_hosts)| union(g_new_master_hosts) | default([]) }}: an unexpected type error occurred. Error was unsupported operand type(s) for +: 'set' and 'unicode'

FATAL: all hosts have already failed -- aborting

**Comment by jprovaznik** [comment]
https://github.com/openshift/openshift-ansible/issues/1865

File: env_origin.yaml [hunk 3]
```diff
@@ -19,4 +19,5 @@ parameters:
   openshift_ansible_git_rev: master
 
 resource_registry:
-    OOShift::LoadBalancer: loadbalancer_neutron.yaml
+  OOShift::LoadBalancer: loadbalancer_neutron.yaml
+  OOShift::ContainerPort: sdn_openshift_sdn.yaml
```
Comment (jprovaznik) [inline env_origin.yaml:23]: Could you please add the same lines to env_aop.yaml too?

Comment (jprovaznik) [inline env_origin.yaml]: Hm, now I see loadbalance ris missing there too, so ignore this - we can add them in a separate patch.

**Comment by jprovaznik** [comment]
As agreed during our last chat, the "oadm diagnostics" error should not occur anymore (albeit no change has been done in this patch, maybe openshift-ansible fix?), merging this as it is.

**Pull request merged**